src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceSettingsClassComposer.java

```diff
@@ -370,13 +372,22 @@ private List<MethodDefinition> createDefaultGetterMethods(Service service, TypeS
         getTransportContext().defaultTransportProviderBuilderNames().iterator();
     Iterator<Class<?>> channelProviderClassesIt =
         getTransportContext().instantiatingChannelProviderBuilderClasses().iterator();
+    boolean secondaryTransportProviderBuilder = false;
     while (providerBuilderNamesIt.hasNext() && channelProviderClassesIt.hasNext()) {
+      List<AnnotationNode> annotations = ImmutableList.of();
```

Comment:
If I'm understanding correctly, this part is saying that we don't generate `BetaApi` annotation for the first getter, but generate it from the second one. I understand we may only have GRPC and HttpJson for now, but it's not easy to read and looks like a hack to me. Can we add the annotation based on something more concrete? Like override this method or part of the method in the subclass?

Reply:
This is an already present paradigm in the TransportContext - it has a bunch of transport-specific lists, where first element in transport is a primary one, all others are secondary. For now there are yes, just two transports, but this is a current architecture decision which I don't think are to be changed in scope of this PR. Overriding method is a valid thing but in that case TransportContext should not exist at all in that case (the difference between transports should be done via overriding methods in each transport). I never liked `TransportContext` as a design decision (and originally it all was done via overriding methods), but i is the current design.

Reply:
I just checked how TransportContext is used and I see where you are coming from. Without changing the design, I think there is still something we can do to improve it:

1. Extract the hardcoded method names as constants, check if the method name matches HttpJson method names, and add BetaApi annotation accordingly. The current logic depends on the order of hardcoded method names which could be easily messed up with.
2.  Add some comments to explain this logic. I know it maybe all in your head and looks simple to you, but with more people starting to maintain this repo, it could benefit us tremendously.

Reply:
Added a comment.
For the hardcoded method names - changing that is beyond the scope of this PR. There are way too many inline string constants here and there, so changing a few here won't change much. Also to the best of my knowledge, having inline constants is idiomatic an preferred to having separate explicit constants if they remain private and are not reused. I think about it this way: we are generating code here, that code has lots of names, having htem as a bunch of constant scattered all over the place does not seem helping but instead making things harder because it adds another level of indirection when inspecting code. Another example: if "printf" style is used for generation then all of those method are also a bunch of strings. It templates is used, then those method names are again just string in templates.

Reply:
I agree we should just used the string literal if it is not reused, I was suggesting more like extracting all the method names and centralize them, then reuse them whenever we need it, and this is a good place for us to reuse it, instead of relying on the order of method names in the list. However, I think this refactoring effort can also be done in a separate PR, it does not have to block this PR.

Reply:
Yes, I think it is a totally valid point and would be probably a good architectural decision, but looks like it would make sense only as a separate global refactoring (which will create that centralized naming repoisitory). Before that "repository" is created, we can't do it properly in scope of this PR.